src/nwb/NWBFile.cpp

```diff
@@ -107,27 +110,33 @@ Status NWBFile::createElectricalSeries(
     return Status::Failure;
   }
 
-  // store all recorded data in the acquisition group
-  std::string rootPath = "/acquisition/";
-
-  // Setup electrode table
-  ElectrodeTable elecTable = ElectrodeTable(io);
-  elecTable.initialize();
+  // Setup electrode table if it was not yet created
+  bool electrodeTableCreated =
+      io->objectExists(ElectrodeTable::electrodeTablePath);
+  if (!electrodeTableCreated) {
+    elecTable = std::make_unique<ElectrodeTable>(io);
+    elecTable->initialize();
+  }
 
   // Create continuous datasets
   for (const auto& channelVector : recordingArrays) {
     // Setup electrodes and devices
     std::string groupName = channelVector[0].groupName;
-    std::string devicePath = "/general/devices/" + groupName;
-    std::string electrodePath = "/general/extracellular_ephys/" + groupName;
-    std::string electricalSeriesPath = rootPath + groupName;
+    std::string sourceName = channelVector[0].sourceName;
```

Comment:
One thing I realized when adding tests for a file that has both `ElectricalSeries` and `SpikeEventSeries` was that we needed to have a way to specify that data came from the same source (e.g. the same device) but had different names within the acquisition group. I added this `sourceName` as a separate input to the `Channel` class as a way to accommodate this.

Reply:
Is ``sourceName`` the same as the name of the ``device``?

Reply:
This is now updated so that `groupName` is used for the ElectrodeGroup and Device and is pulled from the `channelVector` input. The name of the dataset within the `acquisition` group is set by a `recordingArrayNames` input to `createElectricalSeries` and `createSpikeEventSeries`.